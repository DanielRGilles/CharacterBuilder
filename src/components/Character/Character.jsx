import React from 'react'

export default function Character({head, middle, leg}) {
    return (
        <section style={{display: 'flex', flexDirection: 'column'}}>
            <article   style={{
            backgroundImage: `url(./${head}-head.png)`,
            height: 150,
             width: 150,
             backgroundSize: 'cover',
             backgroundRepeat: 'no-repeat',
                 }}>
            </article>
            <article style={{
          backgroundImage: `url(./${middle}-middle.png)`,
          height: 150,
          width: 150,
          backgroundSize: 'cover',
          backgroundRepeat: 'no-repeat',
                }}>  
            </article>

            <article style={{
              backgroundImage: `url(./${leg}-legs.png)`,
              height: 150,
              width: 150,
              backgroundSize: 'cover',
              backgroundRepeat: 'no-repeat',
                }}>
            </article>

        </section>
            
        
    )
}
