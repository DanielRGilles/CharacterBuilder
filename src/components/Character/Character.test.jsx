import {screen, render} from '@testing-library/react';
import Character from './Character';

it('should display the Character component', () => {
  const {container} = render(<Character head='fox' middle='wings' leg='fish' />);

  const head = screen.getByLabelText(/head/);
  const middle = screen.getByLabelText(/middle/);
  const leg = screen.getByLabelText(/leg/);
  expect(head).toHaveStyle(`background-image: url(./fox-head.png)`);
  expect(middle).toHaveStyle(`background-image: url(./wings-middle.png)`);
  expect(leg).toHaveStyle(`background-image: url(./fish-legs.png)`);

  expect(container).toMatchSnapshot();
});