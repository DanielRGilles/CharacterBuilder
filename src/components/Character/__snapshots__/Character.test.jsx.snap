// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display the Character component 1`] = `
<div>
  <section
    style="display: flex; flex-direction: column;"
  >
    <article
      aria-label="head"
      style="background-image: url(./fox-head.png); height: 150px; width: 150px; background-size: cover; background-repeat: no-repeat;"
    />
    <article
      aria-label="middle"
      style="background-image: url(./wings-middle.png); height: 150px; width: 150px; background-size: cover; background-repeat: no-repeat;"
    />
    <article
      aria-label="leg"
      style="background-image: url(./fish-legs.png); height: 150px; width: 150px; background-size: cover; background-repeat: no-repeat;"
    />
  </section>
</div>
`;
